pkg/controller/controllerutil/controllerutil.go

```diff
@@ -248,19 +247,8 @@ func AddFinalizer(o metav1.Object, finalizer string) {
 	o.SetFinalizers(append(f, finalizer))
 }
 
-// AddFinalizerWithError tries to convert a runtime object to a metav1 object and add the provided finalizer.
```

Comment:
hi @dmvolod I think it would be better for now to leave the `WithError` funcs and just add a comment that they are deprecated and shouldn't be used. We can remove them when we also merge other braking changes. AFAIK right now we do not have any breaking changes planned.

/ok-to-test

Reply:
Thanks for feedback, @alvaroaleman 
I've changed just common functions and marked *WithError as deprecated